pkg/addons/addons_gcpauth.go

```diff
@@ -80,6 +80,10 @@ func enableAddonGCPAuth(cfg *config.ClusterConfig) error {
 					creds, _ = google.CredentialsFromJSON(ctx, f)
 				}
 			}
+		} else if !Force && detect.IsOnGCE() && os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") == "" {
```

Comment:
@sharifelgamal sorry, after testing this new change I see that the `return nil` is wrong. If the secret `gcp-auth` is not created then pulling images does not work.

Unless I'm missing some context, I would like to give it another try.

Reply:
Yeah, I think your use-case is that creds.JSON is nil (but getting the credentials doesn't return an error). We should handle adding the pull secret in that case (as long as creds.TokenSource is not nil). I think the best way to do that would be to change where the createPullSecret call gets made.

Instead of `if err != nil || creds.JSON == ""`, we should change that statement to `if err != nil`

then we can create the pull secret

then we can check if creds.JSON is empty after the call to createPullSecret. if it is empty, then return the same error as in the err != nil if statement.

does that make sense?

Reply:
I took a slightly different route and basically checked if `GOOGLE_APPLICATION_CREDENTIALS` is set and `--force` was present, if that is the case then `google.FindDefaultCredentials` will use it.

If that fails, then we do the `detect.IsCloudShell` part and finally if everything else failed and `GOOGLE_APPLICATION_CREDENTIALS` is not set and `detect.IsOnGCE` is true then we just show a warning that the attached service account will be used and continue.

This are my tests cases:

- **With service account**

```
ubuntu@lb-minikube-dev-test:~/minikube$ minikube addons enable gcp-auth
! It seems that you are running in GCE, we will use the attached service account.
* The 'gcp-auth' addon is enabled
```

- **With an empty credentials file (Just to test potential failure) without force**

```
ubuntu@lb-minikube-dev-test:~/minikube$ env GOOGLE_APPLICATION_CREDENTIALS="/home/ubuntu/gcp.json" minikube addons enable gcp-auth

X Exiting due to MK_USAGE: It seems that you are running in GCE and have GOOGLE_APPLICATION_CREDENTIALS environment variable set, if you would still like to authenticate using a credentials file use the --force flag.
```

- **With an empty credentials file (Just to test potential failure) without force**

```
ubuntu@lb-minikube-dev-test:~/minikube$ env GOOGLE_APPLICATION_CREDENTIALS="/home/ubuntu/gcp.json" minikube addons enable gcp-auth --force

* Exiting due to MK_CREDENTIALS_NOT_FOUND: Could not find any GCP credentials. Either run `gcloud auth application-default login` or set the GOOGLE_APPLICATION_CREDENTIALS environment variable to the path of your credentials file.
```

- **With a valid credentials file**

```
ubuntu@lb-minikube-dev-test:~/minikube$ env GOOGLE_APPLICATION_CREDENTIALS="/home/ubuntu/gcp.json" minikube addons enable gcp-auth --force
  - Using image k8s.gcr.io/ingress-nginx/kube-webhook-certgen:v1.0
  - Using image gcr.io/k8s-minikube/gcp-auth-webhook:v0.0.7
* Verifying gcp-auth addon...
* Your GCP credentials will now be mounted into every pod created in the minikube cluster.
* If you don't want your credentials mounted into a specific pod, add a label with the `gcp-auth-skip-secret` key to your pod configuration.
* If you want existing pods to be mounted with credentials, either recreate them or rerun addons enable with --refresh.
```